test/bh/mix-test.js

```diff
@@ -0,0 +1,132 @@
+var fixtures = require('../fixtures')('bemhtml');
+var compile = fixtures.compile;
+
+describe('mix()(×)', function() {
+  it('should return mix', function() {
+    compile(function() {
+      block('button').def()(function() {
+        return JSON.stringify(this.ctx.mix);
+      });
+    })
+      .apply({ block: 'button', mix: { block: 'mix' } })
+      .should.equal('{"block":"mix"}');
+  });
+
+  it('should set single mix', function() {
+    compile(function() {
+      block('button').mix()({ block: 'mix' })
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  it('should set single mix as function', function() {
+    compile(function() {
+      block('button').mix()(function() { return { block: 'mix' }; });
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  it('should set array mix', function() {
+    compile(function() {
+      block('button').mix()([ { block: 'mix' } ]);
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  it('should set array mix as function', function() {
+    compile(function() {
+      block('button').mix()(function() { return [ { block: 'mix' } ]; });
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  xit('should extend user single mix', function() {
+    compile(function() {
+      block('button').mix()({ block: 'mix2' });
+    })
+      .apply({ block: 'button', mix: { block: 'mix1' } })
+      .should.equal('<div class="button mix1 mix2"></div>');
+  });
+
+  xit('should extend user array mix', function() {
+    compile(function() {
+      block('button')(
+        mix()([ { block: 'mix' } ])
```

Comment:
[Bad practice](https://github.com/bem/bem-xjst/wiki/Notable-changes-between-bem-xjst@1.x-and-bem-xjst@2.x#user-content-static-objects-shortcuts-in-mix-content-etc).

[Demo](http://bem.github.io/bem-xjst/?bemhtml=%2F%2F%20Don%27t%20do%20that!%0Ablock(%27b1%27)(%0A%20%20mix()(%5B%5D)%2C%20%2F%2F%20Use%20function%20generator%20instead%3A%20%60function()%20%7B%20return%20%5B%5D%3B%20%7D%60.%0A%20%20mix()(function%20()%20%7B%0A%20%20%20%20var%20res%20%3D%20applyNext()%3B%0A%20%20%20%20res.push(%7Bblock%3A%20%27b2%27%7D)%3B%0A%20%20%20%20return%20res%3B%0A%20%20%7D)%2C%0A%0A%20%20attrs()(%5B%5D)%2C%20%2F%2F%20Use%20function%20generator%20instead%3A%20%60function()%20%7B%20return%20%7B%7D%3B%20%7D%60.%0A%20%20mod(%27a%27%2C%20true).attrs()(function%20()%20%7B%0A%20%20%20%20var%20res%20%3D%20applyNext()%3B%0A%20%20%20%20res.withMod%20%3D%20%27a%27%3B%0A%20%20%20%20return%20res%3B%0A%20%20%7D)%0A)%3B&bemjson=(%5B%0A%20%20%20%20%7B%20block%3A%20%27b1%27%2C%20mods%3A%20%7B%20a%3A%20true%20%7D%20%7D%2C%0A%20%20%20%20%7B%20block%3A%20%27b1%27%20%7D%2C%0A%20%20%20%20%7B%20block%3A%20%27b1%27%20%7D%2C%0A%20%20%20%20%7B%20block%3A%20%27b1%27%20%7D%2C%0A%20%20%20%20%7B%20block%3A%20%27b1%27%20%7D%2C%0A%20%20%20%20%7B%20block%3A%20%27b1%27%20%7D%2C%0A%20%20%20%20%7B%20block%3A%20%27b1%27%20%7D%2C%0A%5D)%3B) :)